File: src/types/content-type.ts
export interface ContentType {
  id: string;
  name: string;
  icon?: string;
  color?: string;
  isBuiltIn: boolean;
  sortOrder: number;
  createdAt: string;
}

export interface NewContentType {
  name: string;
  icon?: string;
  color?: string;
  isBuiltIn: boolean;
  sortOrder: number;
}

File: src-tauri/src/models/content_type.rs
use serde::{Deserialize, Serialize};
use sqlx::prelude::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
#[allow(dead_code)]
pub struct ContentType {
    pub id: String,
    pub name: String,
    pub icon: Option<String>,
    pub color: Option<String>,
    #[sqlx(default)]
    pub is_built_in: bool,
    #[sqlx(default)]
    pub sort_order: i32,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(dead_code)]
pub struct NewContentType {
    pub name: String,
    pub icon: Option<String>,
    pub color: Option<String>,
    pub is_built_in: bool,
    pub sort_order: i32,
}